XenAdmin/ConsoleView/XSVNCScreen.cs

```diff
@@ -295,8 +295,6 @@ private void PollRDPPort(Object Sender)
         {
             if (hasRDP)
             {
-                if (connectionPoller != null)

```

Comment:
Is it necessary to leave the connectionPoller ticking? StartPolling does get called when the VM guest metrics change, perhaps we should be watching for property `other` additionally to networks? After removing the `if (hasRDP)` from line 776, I would think that now the method PollRDPPort will be called because a new connectionPoller will be created (new creation may not be necessary, but this is a different issue - I'm trying out some changes on the side to address this in a subsequent PR).

Reply:
The `connectionPoller` keeps ticking because a user can uninstall the VM tools. This would disable the capability of using RDP, but the button would still show. From my testing, XC wouldn't actually spot this change (or at least not reliably), because the guest metrics change event is fired only when XC tries to read it. So if XC isn't reading the a VM guest tools value, it doesn't know that anything has changed (no event is fired). Been a while since I opened this PR so the details might be slightly different.

Currently, if the VM tools are uninstalled and the user clicks on the RDP button, the user would be shown nothing.
To replicate (using `master`)
1. Install VM tools and reboot
2. Disconnect -> reconnect (this forces XC to read `guest_metrics` and similar)
3. Uninstall VM tools
4. Should be able to press that button

We could address this by try/catching the error, and possibly updating the RDP button to disabled (or hide it?) if pressing it fails. I think it fails (silently) when it looks for IPs in `networks`, and doesn't find them. We could expand that code to alert the user.
This would check for RDP capability every time the button is pressed, which would be used to alert the user that they cannot use RDP anymore.